pkg/lib/authn/sso/oauth_provider.go

```diff
@@ -42,13 +43,26 @@ func (p *OAuthProviderFactory) getActiveOrDemoProvider(alias string) (provider o
 	}
 
 	if config.OAuthSSOProviderConfig(providerConfig).IsMissingCredentialAllowed() {
-		// TODO(tung): handle demo status
+		if p.SSOOAuthDemoCredentials == nil {
+			err = newOAuthProviderMissingCredentialsError(alias, providerConfig.Type(), false)
+			return
+		}
+
+		demoItem, ok := p.SSOOAuthDemoCredentials.LookupByProviderType(providerConfig.Type())
+		if !ok {
+			err = newOAuthProviderMissingCredentialsError(alias, providerConfig.Type(), true)
+			return
+		}
 
-		details := apierrors.Details{
-			"OAuthProviderAlias": alias,
-			"OAuthProviderType":  providerConfig.Type(),
+		deps = &oauthrelyingparty.Dependencies{
+			Clock:          p.Clock,
+			ProviderConfig: demoItem.ProviderConfig,
+			ClientSecret:   demoItem.ClientSecret,
+			HTTPClient:     p.HTTPClient.Client,
+			SimpleStore:    p.SimpleStoreRedisFactory.GetStoreByProvider(providerConfig.Type(), alias),
```

Comment:
providerConfig should be `demoItem.ProviderConfig`?

Reply:
The type should be same, so they are same. Let me still update to use demoItem so we don't have to think about this.